Jint/Key.cs

```diff
@@ -14,7 +15,7 @@ namespace Jint
         private Key(string name)
         {
             Name = name;
-            HashCode = StringComparer.Ordinal.GetHashCode(name);
+            HashCode = Hash.GetFNVHashCode(name);
```

Comment:
Should it be lazy or is any Key always checked at least once for equality?

Reply:
The keys basically always used against dictionaries, we are trying to find identifiers in context (variables, object fields).

Reply:
A stupid idea you will debunk for sure: Assuming most keys are smallish, can't we use an optimization by using the raw memory value of the strings < 8 chars (64bits -> `int`) as the hashcode? And then fallback above 8 chars? This way for <8 chars we just need to compare the hashes too, and the computation could be faster (TBD).

Reply:
Not a stupid idea for sure, makes sense but I'm not an entropy expert (how that would affect collisions). 

I also played with the idea of having length < 4 directly interned and then reference equals would suffice in Key context.

I hope to create a new baseline to improve upon, but feels that perf runs can be quite unreliable - results vary a lot...